corehq/apps/cloudcare/static/cloudcare/js/backbone/apps.js

```diff
@@ -677,6 +678,28 @@ cloudCare.AppView = Backbone.View.extend({
         data.onload = function (adapter, resp) {
             cloudCare.dispatch.trigger("form:ready", form, caseModel);
         };
+        data.answerCallback = function(sessionId) {
+
+            var rawBaseUrl = self.options.urlRoot;
+            var lastIndex = self.options.urlRoot.lastIndexOf('/apps/');
+            var baseUrl = rawBaseUrl.substring(0, lastIndex);
+
+            var render_xml_url = baseUrl + "/render_xml/" + sessionId
```

Comment:
at a minimum i think it'd be nicer to define these as functions. so like `getRenderUrl(baseUrl, sessionId)` just to keep the sketchiness in a localized place